Initial test set-up for datastore-entities

Billcountry opened the pull request:
- Added circle CI config
- Added Codecov config
- Added tests for reading operations including querying and getting
- Added rules for merging to master

cymox1 commented:
# [Codecov](https://codecov.io/gh/echomobi/echo-datastore-entities/pull/1?src=pr&el=h1) Report
> :exclamation: No coverage uploaded for pull request base (`master@b5708c5`). [Click here to learn what that means](https://docs.codecov.io/docs/error-reference#section-missing-base-commit).
> The diff coverage is `n/a`.

[![Impacted file tree graph](https://codecov.io/gh/echomobi/echo-datastore-entities/pull/1/graphs/tree.svg?width=650&height=150&src=pr&token=wISZIbmJbd)](https://codecov.io/gh/echomobi/echo-datastore-entities/pull/1?src=pr&el=tree)

```diff
@@            Coverage Diff            @@
##             master       #1   +/-   ##
=========================================
  Coverage          ?   42.72%           
=========================================
  Files             ?        4           
  Lines             ?      110           
  Branches          ?        0           
=========================================
  Hits              ?       47           
  Misses            ?       63           
  Partials          ?        0           
```



------

[Continue to review full report at Codecov](https://codecov.io/gh/echomobi/echo-datastore-entities/pull/1?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/echomobi/echo-datastore-entities/pull/1?src=pr&el=footer). Last update [b5708c5...d358c8c](https://codecov.io/gh/echomobi/echo-datastore-entities/pull/1?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

georgemarrows commented on tests/test_query.py:
```diff
@@ -0,0 +1,86 @@
+        entities = TestEntity.query().gte("prop2", 20).lt("prop2", 30).fetch()
+        self.assertEqual(len(entities), 10)
+        for entity in entities:
+            self.assertTrue(20 <= entity.prop2 < 30)
```
I didn't know you could do `20 <= entity.prop2 < 30` in Python! Cool.

georgemarrows commented on tests/test_entity.py:
```diff
@@ -0,0 +1,48 @@
+        self.assertEqual(expected_key.id, entity.key().id)
+        # TODO: Test for key generation after put
+
+    def test_get(self):
```
Oh hold on, looks like wrapper's `put` doesn't work yet.

Billcountry replied:
It's a todo coz the library has no write capabilities at the moment. Just read

Billcountry commented on tests/test_entity.py:
```diff
@@ -0,0 +1,48 @@
+
+    def test_value_setting(self):
+        entity = TestEntity()
+        entity.prop1 = "Text Value"
```
I'll add one. Should throw invalid value error.

Billcountry replied:
Done

Billcountry commented on tests/test_entity.py:
```diff
@@ -0,0 +1,48 @@
+        self.assertEqual(str(entity.key()), expected_key.to_legacy_urlsafe().decode("utf-8"))
+        self.assertEqual(entity.key().id, 10)
+        self.assertEqual(expected_key.id, entity.key().id)
+        # TODO: Test for key generation after put
```
Once we have implemented write into the library

georgemarrows replied:
Agree. I saw later that `put` isn't yet there.

Billcountry commented on tests/test_entity.py:
```diff
@@ -0,0 +1,48 @@
+    def test_key_generation(self):
+        entity = TestEntity()
+        # You should not get a key for an unsaved entity
+        self.assertRaises(errors.NotSavedException, entity.key)
```
Nope, `assertRaises` automatically calls the method. Variables are provided as `*args, **kwargs`

georgemarrows replied:
Ah, wasn't thinking how `assertRaises` had to work.

Billcountry commented on tests/test_query.py:
```diff
@@ -0,0 +1,86 @@
+            count += 1
+
+    def test_order_and_limit(self):
+        limit = randint(2, 20)
```
Setting a fixed limit for now

Billcountry commented:
@georgemarrows I've removed stub methods for put.
Added more docs for the methods. I'm not sure we have a way to test `eventual` at the moment.
Key_only is now tested

georgemarrows commented on echo/datastore/entity.py:
```diff
@@ -12,45 +25,103 @@ class Entity(object):
 
-    def key(self):
+    def key(self, partial=False):
+        """Generates a key for this Entity
```
Thanks for adding these comments. Are they taken from the actual Google old datastore API implementation?

Billcountry replied:
Not this one, but some that translate directly yes. e.g. in those in Query
